components/range_cache_memory_engine/src/range_manager.rs

```diff
@@ -71,124 +129,185 @@ impl RangeMeta {
         self.safe_point = safe_point;
     }
 
-    fn derive_from(id: u64, r: &RangeMeta) -> Self {
-        Self {
-            id,
-            range_snapshot_list: SnapshotList::default(),
-            safe_point: r.safe_point,
-        }
+    pub(crate) fn get_range(&self) -> &CacheRange {
+        &self.range
     }
 
-    pub(crate) fn range_snapshot_list(&self) -> &SnapshotList {
-        &self.range_snapshot_list
+    pub fn get_state(&self) -> RegionState {
+        self.state
     }
-}
 
-#[derive(Default)]
-struct IdAllocator(u64);
+    pub(crate) fn set_state(&mut self, new_state: RegionState) {
+        self.state = new_state;
+    }
+
+    pub(crate) fn mark_evict(&mut self, state: RegionState, reason: EvictReason) {
+        use RegionState::*;
+        assert!(matches!(self.state, Loading | Active | LoadingCanceled));
+        assert!(matches!(state, PendingEvict | Evicting));
+        self.state = state;
+        self.evict_info = Some(EvictInfo {
+            start: Instant::now_coarse(),
+            reason,
+        });
+    }
 
-impl IdAllocator {
-    fn allocate_id(&mut self) -> u64 {
-        self.0 += 1;
-        self.0
+    pub(crate) fn set_in_gc(&mut self, in_gc: bool) {
+        assert!(self.in_gc != in_gc);
+        self.in_gc = in_gc;
+    }
+
+    pub(crate) fn is_in_gc(&self) -> bool {
+        self.in_gc
+    }
+
+    pub(crate) fn dervie_from(region: Region, source_meta: &Self) -> Self {
```

Comment:
Please add comments for this function, like when this function will be called (only when split?).

Reply:
```suggestion
    pub(crate) fn derive_from(region: Region, source_meta: &Self) -> Self {
```